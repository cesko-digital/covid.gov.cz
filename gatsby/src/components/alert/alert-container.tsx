import React from 'react';
import Alert from './alert';

// TODO: Add real dataset

interface IProps {
  alerts?: INotice[];
}

interface INotice {
  content: string;
  from: Date;
  to?: Date;
  link?: string;
}

const alerts: INotice[] = [
  {
    content: 'Nemocnice kolabují jako funkce ministra zdravotnictví!',
    from: new Date(),
    link: '#',
  },
];

const isDateBetween = (from: Date, to?: Date): boolean => {
  const now = new Date();
  return (from <= now && now <= to) || (from <= now && !to);
};

const AlertContainer: React.FC<IProps> = () => {
  const active = alerts.filter(({ from, to }) => isDateBetween(from, to));
  return (
    <>
      {active.map((notice, i) => (
        <Alert
          key={`notice-box-${i}`}
          message={notice.content}
          link={notice.link}
          isInfo={false}
        />
      ))}
    </>
  );
};

export default AlertContainer;
