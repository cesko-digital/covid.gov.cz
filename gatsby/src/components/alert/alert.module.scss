@import '../../../assets/scss/design-system/base/variables';

.wrapper {
  background-color: $theme-yellow;
  color: black;
  position: relative;
  z-index: 1;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 8px 24px;

  a {
    color: black;

    &:hover,
    &:focus,
    &:visited {
      color: black;
    }
  }
}
