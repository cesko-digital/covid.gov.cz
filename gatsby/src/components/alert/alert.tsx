import React from 'react';
import Button from '../button';
import GovIcon from '../gov-icon';
import Row from '../row';
import Col from '../col';

interface IProps {
  message: string;
  link?: string;
  isInfo: boolean;
}

const Alert: React.FC<IProps> = ({
  message = 'Probíhá aktualizace dat, prosím, berte na vědomí, že data nemusí být aktuální všude.',
  isInfo,
  link,
}) => {
  return (
    <>
      <Row
        className={
          isInfo
            ? 'bg-primary w-100 m-0 text-white'
            : 'bg-warning w-100 m-0 text-black'
        }
        justify="center"
        alignItems="center"
      >
        <Col col={11} colMd={9} colLg={6}>
          <div className="d-flex align-items-center justify-content-center py-2">
            <GovIcon icon={isInfo ? 'time' : 'alert'} size={20} />
            <span className="ml-2">{message}</span>
            {link != null ? (
              <Button variant="small-black" text="Více" href={link} />
            ) : (
              <></>
            )}
          </div>
        </Col>
      </Row>
    </>
  );
};

export default Alert;
