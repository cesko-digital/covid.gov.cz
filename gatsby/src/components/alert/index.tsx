import AlertContaner from './alert-container';

export { default as Alert } from './alert';

export { default as AlertContaner } from './alert-container';
