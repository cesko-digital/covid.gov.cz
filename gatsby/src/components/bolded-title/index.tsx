import React from 'react';
import styles from './bolded-title.module.scss';
export interface BoldedTitleProps {
  title: string;
  count?: number;
}

/* global JSX */
export const BoldedTitle: React.FC<BoldedTitleProps> = ({ title, count }) => {
  const splittedTitle: Array<JSX.Element | string> = title.split(/(?= )/g);
  return (
    <>
      {count
        ? splittedTitle.map((item, index) => {
            if (index < count) {
              return (
                <span key={index} className={styles.boldedTitle}>
                  {item}
                </span>
              );
            }

            return item;
          })
        : splittedTitle}
    </>
  );
};
