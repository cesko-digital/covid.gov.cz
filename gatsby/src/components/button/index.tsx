export { default } from './button';
export type { ButtonVariant } from './button';
