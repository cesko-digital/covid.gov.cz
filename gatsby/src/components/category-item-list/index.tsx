export { default } from './category-item-list';
export { default as MeasureAreaList } from './measure-area-list';
export { default as SituationAreaList } from './situation-area-list';
