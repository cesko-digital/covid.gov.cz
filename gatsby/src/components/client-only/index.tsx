export { useHasMounted } from './client-only';
