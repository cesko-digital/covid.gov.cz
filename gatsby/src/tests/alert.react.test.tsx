import { Alert } from '@/components/alert';
import React from 'react';
import renderer from 'react-test-renderer';

test('Alert box renders correctly', () => {
  const component = renderer.create(<Alert message="TestovacÃ­ message" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
