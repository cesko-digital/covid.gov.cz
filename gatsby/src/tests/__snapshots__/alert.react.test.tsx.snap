// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert box renders correctly 1`] = `
<div
  className="row justify-content-center align-items-center w-100 m-0 bg-warning text-black"
>
  <div
    className="col-11 col-md-9 col-lg-6"
  >
    <div
      className="d-flex align-items-center justify-content-center py-1"
    >
      <span
        className="pvs-theme-icon-alert"
        style={
          Object {
            "fontSize": 20,
          }
        }
        title=""
      />
      <span
        className="ml-2"
      >
        Testovac√≠ message
      </span>
    </div>
  </div>
</div>
`;
